update docs to install picker (#492)
commerce-docs/microsite-commerce-storefront: src/content/docs/merchants/storefront-builder/commerce-picker.mdx

---
title: Install the Commerce Picker

> description: Learn how to install the Commerce Picker extension, which lets you select items from your catalog to insert into your content.

Reviewer: ```suggestion
description: Learn how to install the Commerce Picker extension, which lets you select catalog items to insert into your content.
```

---

import { Steps } from '@astrojs/starlight/components';
import Aside from '@components/Aside.astro';
import Diagram from '@components/Diagram.astro';

> The **Commerce Picker** is an extension that allows you to select products, categories, and product variants from your catalog and insert them into your content. It provides a visual interface for browsing and selecting commerce items within your authoring environment.

Reviewer: ```suggestion
The Commerce Picker is an extension that lets you select products, categories, and product variants from your catalog and insert them into your content. It provides a visual interface for browsing and selecting commerce items in your authoring environment.
```

## Installing the Commerce Picker

> The Commerce Picker can be installed for both the Visual Editor and Document Authoring tool. Choose the installation method based on your preferred authoring environment.

Reviewer: ```suggestion
The Commerce Picker can be installed for both the Visual Editor and Document Authoring tool. Choose the installation method based on your authoring environment.
```

### For the Visual Editor

> Go to the [Universal Editor Product Picker documentation](https://developer.adobe.com/uix/docs/extension-manager/extension-developed-by-adobe/ue-product-picker/). Complete the setup process as described in the documentation.

Reviewer: ```suggestion
Go to the [Universal Editor Product Picker documentation](https://developer.adobe.com/uix/docs/extension-manager/extension-developed-by-adobe/ue-product-picker/) and complete the setup process.
```

### For the Document Authoring Tool

<Steps>
1. **Access your storefront configuration**: Navigate to your storefront's configuration in Document Authoring by either:
   - Clicking the gear icon in the DA.live browser interface
   - Opening directly: `https://da.live/config#/<your-org>/<your-site>/`

> 2. **Navigate to the Library tab**: Click the "Library" tab in the configuration interface.

Reviewer: ```suggestion
2. **Navigate to the Library tab**: Click the **Library** tab in the configuration interface.
```

3. **Add the Commerce Picker configuration**: Add a new row with the following values:

   <Diagram caption="Commerce Picker Configuration">
     ![Commerce Picker Configuration](images/picker-config.png)
   </Diagram>

   | Field | Value |
   |-------|-------|
   | **Title** | Commerce Picker |
   | **Path** | `https://main--aem-commerce-picker-dist--oneadobe.aem.page/dist/dist/document-authoring.html?configs-path=/config.json` |
   | **Experience** | fullsize-dialog |
   | **Icon** | `https://main--aem-boilerplate-commerce--hlxsites.aem.page/icons/cart.svg` |

> 4. **Update your project configuration**: Modify your project's `config.json` file to set your root category ID under `"plugins.picker.rootCategory"`. Refer to the [demo configuration](https://github.com/hlxsites/aem-boilerplate-commerce/blob/main/demo-config.json#L36).

Reviewer 2: We likely need to add specific steps for ACO, or rework how the picker finds this. For ACCS/PaaS, it may be sufficient to query for `storeConfig.root_category_uid`. But for ACO we will have to do a `categoryTree` call and show the options to the user.

I'm wondering if we can just remove the `rootCategory` prop entirely from the config.json file if we do that.

</Steps>

<Aside type="tip" title="Configuration Path">

> If your configurations are located at a different path than `/config.json`, you can modify the URL query parameter in the path field accordingly.

Reviewer: ```suggestion
If your configurations are at a different path from /config.json, modify the URL query parameter in the path field accordingly.
```
